app/controllers/Simul.scala

```diff
@@ -102,7 +102,11 @@ final class Simul(env: Env) extends LilaController(env) {
   def abort(simulId: String) =
     Open { implicit ctx =>
       AsHost(simulId) { simul =>
-        env.simul.api abort simul.id inject jsonOkResult
+        env.simul.api abort simul.id inject {
+          env.mod.logApi.terminateTournament(ctx.userId.get, simul.fullName)
```

Comment:
never use `Option.get`, it throws an exception on `None`.

Here you can use `simul.hostId` instead.

Reply:
It's not a mod action, why is it logged as such?

Reply:
In the `AsHost` context, I assumed `userId` it's not `None` because it's discarded: https://github.com/ornicar/lila/blob/6948e1d85c138fb4e1d9ca269938fcdf9187f6dd/app/controllers/Simul.scala#L217
The same line also shows that it's not necessarily the host which abort it, but anyone with the `isGranted(_.ManageSimul)` perm, that's why I logged it as a mod action

Reply:
It's never ok to make asumptions based on surrounding code. These asumptions break during refactorings. Instead, we use the type system, and it says that `ctx.userId` can be `None`.

OK for logging, maybe only when `!ctx.userId.contains(simul.hostId)`